src/commissaire/encryption/__init__.py

```diff
@@ -0,0 +1,184 @@
+        :type args: list
+        :param kwargs: All other keyword arguments
+        :type kwargs: dict
+        :returns: str or unicode
+        """
+        return self._b64encode(self._encrypt(
+            data, *args, **kwargs)).decode('utf-8')
+
+    def _encrypt(self, data, *args, **kwargs):
+        """
+        Encryption implementation.
+
+        :param data: The data to decrypt
+        :type data: str or unicode
+        :param args: All other non-keyword arguments
+        :type args: list
+        :param kwargs: All other keyword arguments
+        :type kwargs: dict
+        :returns: bytes
+        """
+        raise NotImplementedError('_encrypt must be overriden')
+
+    def decrypt(self, data, *args, **kwargs):
+        """
+        Decrypts, decodes, and returns the original data.
+
+        :param data: The data to decrypt
+        :type data: str or unicode
+        :param args: All other non-keyword arguments
+        :type args: list
+        :param kwargs: All other keyword arguments
+        :type kwargs: dict
+        :returns: str or unicode
+        """
+        data = self._b64decode(data)
+        return self._decrypt(
+            data, *args, **kwargs).decode('utf-8')
+
+    def _decrypt(self, data, *args, **kwargs):
+        """
+        Decryption implementation.
+
+        :param data: The data to decrypt
+        :type data: str or unicode
+        :param args: All other non-keyword arguments
+        :type args: list
+        :param kwargs: All other keyword arguments
+        :type kwargs: dict
+        :returns: bytes
+        """
+        raise NotImplementedError('_decrypt must be overriden')
+
+
+class GNUPGEncryption(EncryptionBase):
+    """
+    gnupg encryption implementation. If there are no keys already configured
+    this class will generate a new key with the name based on the
+    initialization input.
+    """
+
+    #: The gnupg module
+    import gnupg
+
+    def __init__(self, name, passphrase,
+                 keydir='/etc/commissaire/gpg', verbose=False):
+        """
+        Initializes a new instance of the GNUPG Encryption class.
+
+        :param passphrase: The password to use when decrypting.
+        :type passphrase: str
+        :param name: "Real name" of the key. Used to find generated keys.
+        :type name: str
+        :param keydir: Full path to where the key data will be stored.
+        :type keydir: str
+        :param verbose: If the encryptor should be verbose (default: False)
+        :type verbose: bool
+        """
+        self.__passphrase = passphrase
+        self._encryptor = self.gnupg.GPG(
+            homedir=keydir,
+            verbose=verbose)
+
+        # Use the key if it exists
+        self.fingerprint = None
+        for keydata in self._encryptor.list_keys():
+            for uid in keydata['uids']:
+                if uid.split(' ')[0] == name:
+                    self.fingerprint = keydata['fingerprint']
+                break
```

Comment:
Looping over `keydata['uids']` implies you're allowing for multiple iterations, so should this `break` be indented further?